Upgrade to petsc-3.10.5

fdkong opened the pull request:
Closes #13428

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

fdkong committed:
Upgrade to PETSc-3.10.5 in civet

Refs #13428

fdkong committed:
Fixed dump input files in contact_verification

PETSc-3.10.5 will check if an option is valid

Refs #13428

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/342647/) on 5fb6f2f wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13467/site)

This comment will be updated on new commits.

fdkong committed:
Change the input file to use superlu_dist since strumpack has a bug with 64 bit

    duplicate_fronts=<optimized out>) at /opt/moose/gcc-8.3.0/include/c++/8.3.0/bits/stl_algobase.h:1009
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/arch-linux2-c-opt/externalpackages/git.strumpack/src/MPI_wrapper.hpp:48
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/arch-linux2-c-opt/externalpackages/git.strumpack/src/StrumpackSparseSolverMPIDist.hpp:248
    at /opt/moose/gcc-8.3.0/include/c++/8.3.0/bits/std_function.h:88
    at /opt/moose/gcc-8.3.0/include/c++/8.3.0/bits/std_function.h:297
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/arch-linux2-c-opt/externalpackages/git.strumpack/src/TaskTimer.cpp:79
    name=..., f=..., depth=1) at /opt/moose/gcc-8.3.0/include/c++/8.3.0/bits/std_function.h:260
    at /opt/moose/gcc-8.3.0/include/c++/8.3.0/bits/char_traits.h:322
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/arch-linux2-c-opt/externalpackages/git.strumpack/src/StrumpackSparseSolver.cpp:387
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/src/mat/impls/aij/mpi/strumpack/strumpack.c:441
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/src/mat/interface/matrix.c:3065
    at /tmp/moose_package_build_temp/petsc-default-64-mpich-gcc/petsc-3.10.5/src/ksp/ksp/interface/itfunc.c:723
    at /home/kongf/workhome/rod/moose/scripts/../libmesh/src/solvers/petsc_nonlinear_solver.C:930
    end_norm=@0xfa88d0: 0, relax=false, relaxed_dofs=...) at /home/kongf/workhome/rod/moose/framework/src/executioners/PicardSolve.C:378

moosebuild commented:
Job [Code Coverage](https://civet.inl.gov/job/342734/) on 9055df8 wanted to post the following:

View code coverage site [here](https://mooseframework.inl.gov/docs/PRs/13467/coverage)

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/342758/) on 9055df8 wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/13467/timing)

This comment will be updated on new commits.

fdkong commented:
@milljm @permcody I have done the tests for the whole world. It is safe to merge it in right now.

permcody commented:
Well this is great but @lindsayad also has a libMesh update up. So this is two significant changes that are independently tested but not tested with each other. I'm not super comfortable in merging them both without rebasing one. I guess we can talk about it in the cube since we aren't merging anything just yet.

fdkong commented:
> Well this is great but @lindsayad also has a libMesh update up. So this is two significant changes that are independently tested but not tested with each other. I'm not super comfortable in merging them both without rebasing one. I guess we can talk about it in the cube since we aren't merging anything just yet.

Right way is to hold all PRs now. And merge libMesh or PETSc in, and then test the unmerged one. I definitely do not want to wait forever. We are already too far behind. PETSc has big changes on hypre right now, and I need to check all these important changes ASAP. I so can put some patches back to PETSc-3.11.x.
